distributed/diagnostics/tests/test_worker_plugin.py

```diff
@@ -264,12 +265,14 @@ async def test_assert_no_warning_no_overload(c, s, a):
     class Dummy(WorkerPlugin):
         pass
 
-    with pytest.warns(None):
+    with warnings.catch_warnings(record=True) as record:
```

Comment:
technically `with warnings.catch_warnings(record=True) as record: .... assert not record` isn't needed with `filterwarnings=error` because warnings will throw errors and fail the test. But I left it in in case `filterwarnings` config is ever removed